### PR TITLE
enhancement: add tests general lineplots #287

### PR DESCRIPTION
**Pull request by Gero1999**
Contributes to #287

## Description
This pull request includes updates to the `general_lineplot` function in `R/general_lineplot.R` and its corresponding tests in `tests/testthat/test-general_lineplot.R`. The changes improve functionality, simplify code, and enhance test coverage for edge cases.

### Test Enhancements:
* Fixed a typo in the test data by replacing a duplicate column `ARRLT` with `AFRLT`.
* Added a test to ensure the function returns a `ggplot` object with an appropriate title when the input data is empty.
* Introduced tests to validate the handling of predose records and to ensure correct behavior when `time_scale` is not "By Cycle".

### Other modifications
* Removed the duplicated check for empty data
* Simplified the logic for determining the `time` variable by replacing an `if` statement with a `switch`.

## Definition of Done

- [x] tests have a 100% coverage for `general_lineplot.R` (`covr`)
- [x] tests ARE logical 🧠 

## How to test
- check tests in file `tests/testthat/test-pivot_wider_pknca_results.R`

## Contributor checklist
- [x] Code passes lintr checks
- [x] Code passes all unit tests
- [x] New logic covered by unit tests
- [x] New logic is documented
- [x] Package version is incremented~

## Notes to reviewer

Anything that the reviewer should know before tacking the pull request?

### CODE REVIEW

**Review by js3110 — APPROVED** [review]
LGTM!

File: R/general_lineplot.R [hunk 1]
```diff
@@ -115,11 +110,7 @@ general_lineplot <- function(
       filter(AVAL > 0)
   }
 
-  time <- if (time_scale == "By Cycle") {
-    "ARRLT"
-  } else {
-    "AFRLT"
-  }
+  time <- switch(time_scale, `By Cycle` = "ARRLT", "AFRLT")
```
Comment (m-kolomanski) [inline R/general_lineplot.R:113]: suggestion: This is marginal, but simple `if else` statement fits better here.
```suggestion
  time <- if (time_scale == "By Cycle") "ARRLT" else "AFRLT"
```

**Review by m-kolomanski — APPROVED** [review]
LGTM
